interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/ksTypes.kt

```diff
@@ -56,7 +56,15 @@ public fun KSType.toTypeName(
   typeParamResolver: TypeParameterResolver = TypeParameterResolver.EMPTY
 ): TypeName {
   val type = when (val decl = declaration) {
-    is KSClassDeclaration -> decl.toClassName().parameterizedBy(arguments.map { it.toTypeName(typeParamResolver) })
+    is KSClassDeclaration -> {
+      decl.toClassName().let { cn ->
+        if (arguments.isEmpty()) {
+          cn
+        } else {
+          cn.parameterizedBy(arguments.map { it.toTypeName(typeParamResolver) })
```

Comment:
can this `if` check be merged? since an empty arguments will just replace no arguments in the `ClassName`

Reply:
Unfortunately not, KP will throw an error if you parameterize with an empty list